### ISSUE TITLE
rchk problems

### ISSUE THREAD

**Issue by krlmlr** [cwida/duckdb#1032]
Currently:

```
Package duckdb version 0.2.1-2
Package built using 79279/R 4.1.0; x86_64-pc-linux-gnu; 2020-09-30 14:05:37 UTC; unix   
Checked with rchk version 36c7ad2294619ba0a81109c9acb675eea2c96e6d
More information at https://github.com/kalibera/cran-checks/blob/master/rchk/PROTECT.md

Function duckdb_execute_R
  [UP] unprotected variable <unnamed var:   %2 = alloca %struct.SEXPREC*, align 8> while calling allocating function std::__1::unique_ptr<duckdb::QueryResult, std::__1::default_delete<duckdb::QueryResult> >::~unique_ptr() duckdb/src/duckdbr.cpp:611
```

While this might be a false positive, it would be useful to remove this error. Could the `~StreamQueryResult()` destructor allocate (theoretically) through some obscure path?

I see two remedies:

1. Switch to `unique_ptr<MaterializedQueryResult>`, perhaps through a helper function. (Let's hope this destructor isn't flagged as an allocating function.)
2. Introduce a helper function that stores the intended result in a caller-owned `VECSXP` object of length one.

It is unfortunate that we can't seem to easily replicate locally what CRAN is seeing: https://github.com/kalibera/rchk/issues/24.

**Comment by hannesmuehleisen** [comment]
Can you try `generic_result.reset();` just before the last `UNPROTECT` in  `duckdb_execute_R` ? I don't have rchk running here :/

**Comment by krlmlr** [comment]
I can't replicate the results either: https://github.com/kalibera/rchk/issues/24.

**Comment by krlmlr** [comment]
Merged now in #1041.
